pkg/common/quotas/quotas.go

```diff
@@ -148,36 +153,50 @@ func (q *QuotaChecker) getNamespaceOverrides(ctx context.Context, c client.Clien
 
 // AddInitalNamespaceQuotas will add the default quotas to the namespace annotations.
 // It will only add them, if there are currently no such annotations in place.
-func AddInitalNamespaceQuotas(ns *corev1.Namespace) bool {
+func AddInitalNamespaceQuotas(ctx context.Context, iof *iofruntime.Runtime, ns *corev1.Namespace, objMeta *metaonly.MetadataOnlyObject) bool {
 	annotations := ns.GetAnnotations()
 	if annotations == nil {
 		annotations = map[string]string{}
 	}
 
 	added := false
 
+	r := &utils.Resources{
+		CPURequests:    *utils.DefaultCPURequests,
+		CPULimits:      *utils.DefaultCPULimits,
+		MemoryRequests: *utils.DefaultMemoryRequests,
+		MemoryLimits:   *utils.DefaultMemoryLimits,
+	}
+
+	if objMeta.Kind == "XVSHNPostgreSQL" {
+		comp := vshnv1.VSHNPostgreSQL{}
```

Comment:
You could put this logic into its own function, then `AddInitalNamespaceQuotas` will be a bit shorter

Reply:
Extracted that logic into a separte function